controllers/idler/predicate.go

```diff
@@ -0,0 +1,52 @@
+package idler
+
+import (
+	"github.com/codeready-toolchain/member-operator/pkg/webhook/mutatingwebhook"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	runtimeevent "sigs.k8s.io/controller-runtime/pkg/event"
+)
+
+type PodIdlerPredicate struct {
+}
+
+// Update triggers reconcile if the pod runs in users namespace
+// and if either the highest restart count is higher than the threshold
+// or the startTime was newly set in the new version of the pod
+func (p PodIdlerPredicate) Update(event runtimeevent.UpdateEvent) bool {
+	isUserPod, newPod := isUserPod(event.ObjectNew)
+	if !isUserPod {
+		return false
+	}
+	if oldPod, ok := event.ObjectOld.(*corev1.Pod); ok {
+		startTimeNewlySet := oldPod.Status.StartTime == nil && newPod.Status.StartTime != nil
+		return startTimeNewlySet || getHighestRestartCount(newPod.Status) > RestartThreshold
```

Comment:
Maybe have the `RestartThreshold` as a field of the predicate instead of using the global variable? That'll a) make the code slightly more local and therefore IMHO more readable and b) open it up to be user-configurable.

Reply:
it should use the same value as is used in idler_controller.go. There is nothing to be configurable for now, if we wanted to have it configurable, then it should go into the ToolchainConfig, but this is not really needed.
I unexported the constant to make it clear that it's not a global variable, but specific to the controller. 2001124d